sub_maps:
  ros__parameters:
    pose_graph_dir: "/root/dddmr_bags/benanli_detention_basin_pg"
    sub_map_search_radius: 50.0
    sub_map_warmup_trigger_distance: 20.0
    complete_map_voxel_size: 0.5
mcl_3dl:
  ros__parameters:
    init_x: 0.0
    init_y: 0.0
    init_z: 0.0
    init_roll: 0.0
    init_pitch: 0.0
    init_yaw: 1.57
    init_var_x: 2.0
    init_var_y: 2.0
    init_var_z: 0.5
    init_var_roll: 0.1
    init_var_pitch: 0.1
    init_var_yaw: 0.5
    map_frame: "map"
    robot_frame: "base_link"
    odom_frame: "odom"
    map_downsample_x: 0.1
    map_downsample_y: 0.1
    map_downsample_z: 0.1
    map_update_interval_sec: 2 #int
    num_particles: 60

    resample_var_x: 0.2
    resample_var_y: 0.2
    resample_var_z: 0.2
    resample_var_yaw: 0.1
    resample_var_pitch: 0.2
    resample_var_roll: 0.2
    expansion_var_x: 0.5
    expansion_var_y: 0.5
    expansion_var_z: 0.5
    expansion_var_roll: 0.2
    expansion_var_pitch: 0.2
    expansion_var_yaw: 0.2
    match_ratio_thresh: 0.0

    euc_cluster_distance: 0.8
    euc_cluster_min_size: 3
    euc_cluster_max_size: 2400

    knn_num_of_ground_normals: 20

    update_min_d: 0.1
    update_min_a: 0.1

    odom_err_lin_lin: 0.6
    odom_err_lin_ang: 0.3
    odom_err_ang_lin: 0.3
    odom_err_ang_ang: 0.6

    odom_err_integ_lin_tc: 5.0
    odom_err_integ_ang_tc: 10.0

    lpf_step: 2.0
    acc_lpf_step: 128.0

    jump_dist: 1.0
    jump_ang: 1.57

    bias_var_dist: 2.0
    bias_var_ang: 1.57
    tf_tolerance: 0.1
    publish_tf: true

    likelihood:
      num_points_global: 8
      match_dist_min: 0.3
      match_dist_flat: 0.05
      threshold_for_trusted_ground: 6
      radius_of_ground_search: 1.0
